### PR TITLE
Clear old records for clean sessions

### PR DESCRIPTION
**Pull request by muneebahmed10**
*Description of changes:*
Currently, `MQTT_Connect` does not clear records for a clean session, nor when the CONNACK's session present flag is set to false by the broker. Clients are required by the MQTT spec to discard state for clean sessions, and not clearing state when session present is false could lead to the state records being cluttered with dead publishes.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/src/mqtt.c [hunk 1]
```diff
@@ -1558,25 +1561,39 @@ static MQTTStatus_t receiveConnack( const MQTTContext_t * pContext,
...
+               ( status == MQTTSuccess ) )
+        {
+            status = sendPublishAcks( pContext, packetId, state );
+
+            packetId = MQTT_PubrelToResend( pContext, &cursor, &state );
+        }
+    }
+    else
+    {
+        ( void ) memset( pContext->outgoingPublishRecords,
```
Comment (aggarw13) [inline libraries/standard/mqtt/src/mqtt.c:1591]: A comment stating that publish records are being cleared due to clean session would be good to have

File: libraries/standard/mqtt/src/mqtt.c [hunk 2]
```diff
@@ -277,14 +277,17 @@ static MQTTStatus_t receiveConnack( const MQTTContext_t * pContext,
                                     bool * pSessionPresent );
 
 /**
- * @brief Resends pending acks for a re-established MQTT session.
+ * @brief Resends pending acks for a re-established MQTT session, or
+ * clears existing state records for a clean session.
  *
  * @param[in] pContext Initialized MQTT context.
+ * @param[in] sessionPresent Session present flag from the MQTT broker.
```
Comment (aggarw13) [inline libraries/standard/mqtt/src/mqtt.c:284]:
```suggestion
 * @param[in] sessionPresent Session present flag that is received from the MQTT broker.
```

File: libraries/standard/mqtt/utest/mqtt_utest.c [hunk 3]
```diff
@@ -1033,6 +1034,11 @@ void test_MQTT_Connect_happy_path()
     mqttContext.keepAliveIntervalSec = 0;
     connectInfo.cleanSession = true;
     sessionPresentExpected = false;
+    /* Populate some state records to make sure they are cleared. */
```
Comment (aggarw13) [inline libraries/standard/mqtt/utest/mqtt_utest.c:1037]:
```suggestion
    /* Populate some state records to make sure they are cleared as we are establishing a clean session with the broker. */
```

File: libraries/standard/mqtt/src/mqtt.c (hunk 1 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/src/mqtt.c:1592]: Thanks, I forgot to add one here

**Pull request merged**